subprojects/core/src/main/java/org/gradle/internal/classpath/transforms/AdhocInterceptors.java

```diff
@@ -138,6 +147,25 @@ public boolean visitMethodInsn(MethodVisitorScope mv, String className, int opco
                 return visitINVOKESTATIC(mv, className, owner, name, descriptor);
             case Opcodes.INVOKEVIRTUAL:
                 return visitINVOKEVIRTUAL(mv, className, owner, name, descriptor);
+            case Opcodes.INVOKEINTERFACE:
+                return visitINVOKEINTERFACE(mv, owner, name, descriptor);
+        }
+        return false;
+    }
+
+    private boolean visitINVOKEINTERFACE(MethodVisitorScope mv, String owner, String name, String descriptor) {
+        return interceptGetConvention(mv, owner, name, descriptor);
+    }
+
+    /**
+     * Project.getConvention -> ProjectInterceptors.interceptGetConvention()
+     *
+     * This generates the code placed where Project.getConvention() is called.
+     */
+    private static boolean interceptGetConvention(MethodVisitorScope mv, String owner, String name, String descriptor) {
+        if (PROJECT_TYPE.getInternalName().equals(owner) && "getConvention".equals(name) && RETURN_CONVENTION.equals(descriptor)) {
```

Comment:
❓ Would it make sense to split `Convention` logic to a separate special class `JvmBytecodeCallInterceptor` and then AdhocInterceptors would call it. I had something like this in mind:
```
class AdhocInterceptors {
    public boolean visitMethodInsn(...) {
        if (conventionInterceptors.visitMethodInsn(...)) {
            return;
        }
        switch (opcode) {
        ...
        }
    }
}
```
That way it might be easier to delete this logic after 10 years :D

Reply:
Looks a good solution to me 👍